src/psyclone/psyir/transformations/parallel_loop_trans.py

```diff
@@ -189,16 +196,57 @@ def validate(self, node, options=None):
                        signatures_to_ignore=signatures):
             # The DependencyTools also returns False for things that are
             # not an issue, so we ignore specific messages.
+            errors = []
             for message in dep_tools.get_all_messages():
                 if message.code == DTCode.WARN_SCALAR_WRITTEN_ONCE:
                     continue
-                all_msg_str = "\n".join([str(m) for m in
-                                         dep_tools.get_all_messages()])
-                messages = (f"Loop cannot be parallelised because the "
-                            f"dependency analysis reported:\n{all_msg_str}\n"
+                if (privatise_arrays and
+                        message.code == DTCode.ERROR_WRITE_WRITE_RACE):
+                    privatisable = True
+                    for var_name in message.var_names:
+                        try:
+                            sym = node.scope.symbol_table.lookup(var_name)
+                        except KeyError:
+                            # Structures are reported with the full expression:
+                            # "mystruct%myfield" by the DA var_name, we
+                            # purposely avoid privatising these.
+                            privatisable = False
+                            break
+
+                        # If it's not a local symbol, we cannot safely analyse
+                        # its lifetime
+                        if not isinstance(sym.interface, AutomaticInterface):
+                            privatisable = False
+                            break
+
+                        # It must not be referenced after this loop (before the
+                        # loop is fine because we can use OpenMP/OpenACC
+                        # first-private or Fortran do concurrent local_init())
```

Comment:
lastprivate does not exist in OpenACC or std Fortran or omp teams, it also has strange restrictions in OpenMP, e.g. omp loop says: "A list item may not appear in a lastprivate clause unless it is the loop iteration variable of a loop that is associated with the construct." So I would ignore "lastprivate" completely.

I am not sure what you mean in the second part of your sentence, but this is unrelated to lastprivate. I wanted to say that we can not check if a global is private because we don't know if it is used somewhere else before or after the loop. We can only do this with symbols that only live in the local symbol table.